services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -620,13 +617,29 @@ public void resetPassword(CredentialRepresentation cred) {
     @NoCache
     @Produces(MediaType.APPLICATION_JSON)
     public List<CredentialRepresentation> credentials(){
-        auth.users().requireManage(user);
+        auth.users().requireView(user);
```

Comment:
Yes, I did this intentionally as "requireView" is in general used for getters/read use-cases when "requireManage" is used for writes. This endpoint is for reading, I think it is copy/paste error that it has "requireManage" .

But on the other hand, this change is not directly related to the main point of this PR. So if you prefer, I can remove it and we can address admin REST permissions for later. WDYT?

Reply:
Thanks for the explanation. Does that mean that anyone able to view the user would be able to see credentials this user has?

Reply:
Yes, but just the public metadata of the credentials, not the secret data. So for example he will be able to see that user has password and algorithm "pbkdf2" was used to hash the password and there are 20.000 hashing iterations. He won't be able to see secret data (salt and password hash in case of password).

Reply:
I suggest then leaving this information only available for those managing this user since the metadata can increase the attack surface (like "as an attacker I don't need to try password-breaking since the user has enrolled no password").

Reply:
Yes, ok. This change is not related to this PR, so I reverted it. We can potentially revisit in the future.

Reply:
Thanks! I am totally on board with revisiting this type of rights